File: backend/controllers/auth.controller.js
const authService = require('../services/auth.service');

const register = async (req, res) => {
  try {
    const { username, password, ethAddress } = req.body;
    if (!username || !password || !ethAddress) {
      return res.status(400).json({ message: 'All fields are required.' }); // <-- IZMENA
    }

    const user = await authService.registerUser(username, password, ethAddress);

    res.status(201).json({
      message: 'User registered successfully.', // <-- IZMENA
      user: {
        id: user.id,
        username: user.username,
        ethAddress: user.ethAddress,
      },
    });
  } catch (error) {
    res.status(400).json({ message: error.message });
  }
};

const login = async (req, res) => {
  try {
    const { username, password } = req.body;
    if (!username || !password) {
      return res.status(400).json({ message: 'Username and password are required.' }); // <-- IZMENA
    }
    const result = await authService.loginUser(username, password);
    res.json(result);
  } catch (error) {
    res.status(400).json({ message: error.message });
  }
};

module.exports = { register, login };

File: backend/controllers/auth_controller.go
package controllers

import (
	"net/http"

	"github.com/Majlo123/sugar-beet-backend/services"
	"github.com/gin-gonic/gin"
)

type registerRequest struct {
	Username   string `json:"username"`
	Password   string `json:"password"`
	EthAddress string `json:"ethAddress"`
}

type loginRequest struct {
	Username string `json:"username"`
	Password string `json:"password"`
}

func Register(c *gin.Context) {
	var req registerRequest
	_ = c.ShouldBindJSON(&req)

	if req.Username == "" || req.Password == "" || req.EthAddress == "" {
		c.JSON(http.StatusBadRequest, gin.H{"message": "All fields are required."})
		return
	}

	user, err := services.RegisterUser(req.Username, req.Password, req.EthAddress)
	if err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"message": err.Error()})
		return
	}

	c.JSON(http.StatusCreated, gin.H{
		"message": "User registered successfully.",
		"user": gin.H{
			"id":         user.ID,
			"username":   user.Username,
			"ethAddress": user.EthAddress,
		},
	})
}

func Login(c *gin.Context) {
	var req loginRequest
	_ = c.ShouldBindJSON(&req)

	if req.Username == "" || req.Password == "" {
		c.JSON(http.StatusBadRequest, gin.H{"message": "Username and password are required."})
		return
	}

	token, err := services.LoginUser(req.Username, req.Password)
	if err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"message": err.Error()})
		return
	}

	c.JSON(http.StatusOK, gin.H{"token": token})
}
